Overhaul of documentation

RMichaelSwan opened the pull request:
Working on getting documentation to a state where anyone could pick it up and use HARMONI.

RMichaelSwan committed:
overhaul of documentation organization.

RMichaelSwan commented:
TODO: 

1. current package list and roadmap page (make it into roadmap page and reference package section)
2. move/fix remaining content from misc/usage page.

RMichaelSwan committed:
added core package docs, interface listings.

chrismbirmingham commented on doc/Overview.rst:
```diff
@@ -1,13 +1,19 @@
 
+Human And Robot Modular OpeN Interactions (HARMONI) is a comprehensive tool containing all the components and capabilities you need to quickly get your social interaction up and running on a robot. 
+
+HARMONI packages the additional capabilities needed for social human-robot interaction neatly on top of ROS by wrapping and integrating several state of the art libraries in the domains of natural language understanding, dialog, object and face detection, and decision trees. HARMONI is the glue that integrates these disparate packages on top of ROS in a standardized way so you don't have to. Just ...
```
Is the second link supposed to link to configuration? Configuration.html instead of quickstart/Multi-Container? Or is it actually the container configuration, in which case it should be tutorials/Multi-Container.html

RMichaelSwan replied:
Good catch. Yes, configuration link. Updated.

chrismbirmingham commented:
Do we need to update all the package readmes before closing this? Or would we prefer to leave that for another branch/request?

chrismbirmingham commented on harmoni_detectors/harmoni_stt/README.md:
```diff
@@ -1,4 +1,4 @@
+# harmoni_stt
 
 Note: Run get_w2l_models.sh before attempting to launch this service.
 
```
```suggestion
To test if the model has been built properly off of your microphone you can run the following in the terminal:

`ffmpeg -hide_banner -loglevel error -f alsa -i default -ar 16000 -ac 1 -ab 256k -f wav - | /root/wav2letter/build/inference/inference/examples/simple_streaming_asr_example --input_files_base_path=/root/model/w2l/`

`ffmpeg -hide_banner -loglevel error -f alsa -i hw:3 -ar 16000 -ac 1 -ab 256k -f wav - | /root/wav2letter/build/inference/inference/examples/simple_streaming_asr_example --input_files_base_path=/root/model/w2l/`

https://trac.ffmpeg.org/wiki/Capture/ALSA

or

`roslaunch harmoni_detectors/harmoni_stt/launch/direct_stt_service.launch test:=true`

`ffmpeg -hide_banner -loglevel error -f alsa -i hw:3 -ar 16000 -ac 1 -ab 256k -f wav - | /root/wav2letter/build/inference/inference/examples/simple_streaming_asr_example --input_files_base_path=/root/model/w2l/`
```

chrismbirmingham commented on doc/tutorials/Compose-an-Interaction.md:
```diff
@@ -1,6 +1,8 @@
+![packges](../images/PatternBehaviorTree.png)
 
 _An example interaction script pictured as a behavior tree._
 
```
I would suggest we state explicitly that the javascript example below is a template, and that further down is an actual demo example, so that people are not confused thinking that what they should make looks like the javascript. (Obvious for some, but maybe not for novice users).

RMichaelSwan replied:
Definitely. This was copied from the old tutorial. We probably need to run through many of our tutorials again to make sure they're still good.

chrismbirmingham commented on doc/tutorials/Launching-Services.md:
```diff
@@ -15,7 +15,7 @@ Launching packages this way will allow you to verify if the package is correctly
+Similarly, if there are seperate computers controlling the robot hardware, you may wish to have a seperate image for running on that computer. This is often named 'harmoni_hardware'.
 
 
 ### Roslaunch Aliases
```
If we are including these in the docs they should be tested to make sure they still work (I think some are outdated)

RMichaelSwan replied:
Agreed. I wasn't really sure what to do with it and was unsure about removing things...a lot of the tutorials need a review and possible rework.

chrismbirmingham commented on doc/tutorials/Multi-PC.md:
```diff
@@ -1,8 +1,10 @@
+_This provided example covers the two PC setup that exists on QTrobot, which was an early deployment platform._
 
 Run the following commands. A heading indicating #QTRP or #QTPC will be included each time context changes. Any host or container can be ROS Master as long as the ROS_MASTER_URI is set correctly.
 
```
I think the approaches below might need further explanation. This is actually a super important section for most use cases, and so we want to make it really clear. One thing that might not be immediately clear is what needs to be set where, and what needs to be run on which machine.

RMichaelSwan replied:
Hmm. Maybe I should add a diagram and have clearer separations somehow? Suggestions are welcome. I worked on this with @micolspitale93 a while ago, but should probably verify how she's working with it right now.

micolspitale93 replied:
For Misty, we have created a docker compose which relied on the host network (as for the QT). For QT, I remember I have followed the steps of the docker run mentioned in the MULTI_PC redme.

chrismbirmingham commented on doc/Contributions.md:
```diff
@@ -1,7 +1,5 @@
 # Contributing
 
```
```suggestion

HARMONI is intended to serve the needs of the HRI community. We hope that HARMONI currently supports the basic needs of most HRI developers but we acknowledge there will always be room for new capabilities and improvements. As you use HARMONI for your own work we hope you will contribute back the new capabilities and robots you work with for the good of the whole community.
```

chrismbirmingham commented on doc/packages/Core.rst:
```diff
@@ -0,0 +1,13 @@
+harmoni_core
+================================
+
+Everything needed to make the HARMONI system functional.
```
```suggestion
Everything needed to make the HARMONI packages function together.
```

chrismbirmingham commented:
Is there a reason Contributions.rst and Testing.rst got moved to the root folder?

RMichaelSwan committed:
Update harmoni_detectors/harmoni_stt/README.md

Co-authored-by: Chris Birmingham <cmbirmingham19@gmail.com>

RMichaelSwan committed:
Update doc/Contributions.md

Co-authored-by: Chris Birmingham <cmbirmingham19@gmail.com>

RMichaelSwan committed:
Update doc/packages/Core.rst

Co-authored-by: Chris Birmingham <cmbirmingham19@gmail.com>

RMichaelSwan commented:
> Do we need to update all the package readmes before closing this? Or would we prefer to leave that for another branch/request?

@chrismbirmingham If it's convenient to update a readme, let's do it now, otherwise we can do it on an as needed basis. Most of the docs can be updated without a pull request I think, as long as they don't change code and aren't too drastic.

RMichaelSwan commented:
> Is there a reason Contributions.rst and Testing.rst got moved to the root folder?

@chrismbirmingham I didn't see a strong need for having folders for them as they seem pretty short and having a TOC for a single page doesn't make sense. If that changes, we can always add the folders back in.

RMichaelSwan committed:
doc review fixes.

micolspitale93 commented on doc/configuration/Hardware.md:
````diff
@@ -0,0 +1,64 @@
+rostest harmoni_speaker speaker.test
+```
+If it works, you're good! Otherwise, re-do your [audio config](#audio), and try the above command until the audio works.
+
````
Should we add also a reference to the microphone configuration?
Maybe I have missed it, but nowhere I could find any reference to the `configuration.yaml` file for microphone and camera which edits are fundamentals.

RMichaelSwan replied:
I put in only what I knew of/was existing. If you can add more info either to the microphone package or to the hardware config, that would be great.

micolspitale93 commented on doc/tutorials/Multi-PC.md:
````diff
@@ -213,4 +170,65 @@ rostopic list
\ No newline at end of file
+```
+
+### Troubleshooting
````
I have tried to use the swarm approach also for Misty, but I was not able to make it working (I don't know why), so I went for the host network approach. I think it is worth to try the swarm approach again instead.

RMichaelSwan replied:
I did have it working in my testing a while back...could revisit soon. I don't think that update should be part of this PR though.

micolspitale93 reviewed (commented):
Now HARMONI looks a really well-documented repo!!
Should we add also some quick tutorial to create a bot with Lex with the corresponding value of the configuration.yaml file? The same also for other service (e.g., Dialogflow) which required the interface with external service.

RMichaelSwan commented:
> Now HARMONI looks a really well-documented repo!!
> Should we add also some quick tutorial to create a bot with Lex with the corresponding value of the configuration.yaml file? The same also for other service (e.g., Dialogflow) which required the interface with external service.

Thanks! 

That tutorial would be awesome and highly useful, especially if it touched on a little of the setup on the Lex side too. If my time opens up towards the end of the week, I can maybe look into it, but help would be much appreciated.

micolspitale93 committed:
adding instructions for camera and microphone

Just added more specific instruction to make the camera and microphone working, editing the configuration file. 
The specific packages README still need to be updated

micolspitale93 committed:
updating commands for testing tts and bot

micolspitale93 committed:
fixing typo

missing \. it failed in building the compose

chrismbirmingham committed:
Update READMEs
